File: LeetCode/0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        min_price = 100001
        profit = 0
        
        for price in prices:
            if price - min_price > profit:
                profit = price - min_price
                
            if min_price > price:
                min_price = price
        
        return profit

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int profit = 0;

        int minPrice = 100001;

        for (int price : prices) {
            if (price - minPrice > profit) {
                profit = price - minPrice;
            }

            if (minPrice > price) {
                minPrice = price;
            }
        }

        return profit;
    }
}